float64 angle
---
float64 angle_done
uint32 updates_n
---
float64 angle_left
